apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-inserter-tab-panel-selected.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './track-record-event';
+
+const tabPanelRegex = /tab-panel-\d*-(\w*)/;
+const gutenbergTabPanelName = ( tabPanel ) =>
+	( tabPanel.id.match( tabPanelRegex ) || [ null, null ] )[ 1 ];
+
+/**
+ * Return the event definition object to track `wpcom_block_picker_tab_panel_selected`.
+ *
+ * @returns {{handler: Function, selector: string, type: string}} event object definition.
+ */
+export default () => ( {
+	// It would be nice to filter out events where the tab `is-active` before
+	// the click, but we can't do that because the update has already happened
+	selector: ( e ) => gutenbergTabPanelName( e.target ),
```

Comment:
Any reason why we're passing an anon function here?

Not a blocker, just wondering. It seems superfluous, unless I'm missing something? E.g., 

```
selector: gutenbergTabPanelName,
```

Then 

```
const gutenbergTabPanelName = ( event ) => {
	const { id } = event.target;
	return ( id.match( tabPanelRegex ) || [ null, null ] )[ 1 ];
};
```

Reply:
I didn't consider it very deeply, so I'll have to reconstruct some justification ;)

I do like the separation of concerns. Keeping the `.event` there signals that this selector is operating on dom events (not redux events), and we might theoretically want the name of a tab panel that we got through a parent/child relationship or something (instead of an event).

I'm very weakly opinionated here though - I'll go ahead and deploy as-is, but I'm happy to follow up if you'd like to see them combined.